### ISSUE TITLE
contains_over_first_not_nil fails to warn with nil != first(where:

### ISSUE THREAD

**Issue by phoney** [realm/SwiftLint#3212]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

The contains_over_first_not_nil rule fails to warn when the code is ```nil != first(where:```

##### Complete output when running SwiftLint, including the stack trace and command used

This shows one warning but should show two warnings as in containsOverFirstNotNilBug below.
```bash
~/develop/Samples/TestDebugClosures $ swiftlint lint
Loading configuration from '.swiftlint.yml'
Linting Swift files at paths 
Linting 'ViewController.swift' (1/1)
/Users/me/develop/Samples/TestDebugClosures/TestDebugClosures/ViewController.swift:31:14: warning: Contains over first not nil Violation: Prefer `contains` over `first(where:) != nil` (contains_over_first_not_nil)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?
0.39.2
* Installation method used (Homebrew, CocoaPods, building from source, etc)?
brew
* Paste your configuration file:
included:
  - TestDebugClosures
excluded:
  - TestDebugClosures/AppDelegate.swift
  - TestDebugClosures/SceneDelegate.swift
opt_in_rules:
  - contains_over_first_not_nil

```yml
# insert yaml contents here
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  If so, paste their relative paths and respective contents.
no
* Which Xcode version are you using (check `xcodebuild -version`)?
Xcode 11.4.1
Build version 11E503a

* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
	func containsOverFirstNotNilBug() {
		let numbers = [1, 2, 3, 4, 5]

		//  warning: Contains over first not nil Violation: Prefer `contains` over `first(where:) != nil` (contains_over_first_not_nil)
		let has4 = numbers.first(where: { $0 > 3 }) != nil	// Warning as expected
		// The next line has the same violation but there's no warning
		let has4a = nil != numbers.first(where: { $0 > 3 }) // No Warning

		print("\(has4) \(has4a)")
	}
```

// Shows the contains_over_first_not_nil warning as expected
~ $ echo "let has4a = [1, 2, 3, 4, 5].first(where: { $0 > 3 }) != nil" | swiftlint lint --no-cache --use-stdin --enable-all-rules 
<nopath>:1:13: warning: Contains over first not nil Violation: Prefer `contains` over `first(where:) != nil` (contains_over_first_not_nil)
<nopath>:1:1: warning: Explicit ACL Violation: All declarations should specify Access Control Level keywords explicitly. (explicit_acl)
<nopath>:1:1: warning: Explicit Top Level ACL Violation: Top-level declarations should specify Access Control Level keywords explicitly. (explicit_top_level_acl)
<nopath>:1:5: warning: Prefixed Top-Level Constant Violation: Top-level constants should be prefixed by `k`. (prefixed_toplevel_constant)
<nopath>:1:13: warning: Trailing Closure Violation: Trailing closure syntax should be used whenever possible. (trailing_closure)
Done linting! Found 5 violations, 0 serious in 1 file.

// Fails to show the contains_over_first_not_nil warning
~ $ echo "let has4a = nil != [1, 2, 3, 4, 5].first(where: { $0 > 3 }) " | swiftlint lint --no-cache --use-stdin --enable-all-rules 
<nopath>:1:1: warning: Explicit ACL Violation: All declarations should specify Access Control Level keywords explicitly. (explicit_acl)
<nopath>:1:1: warning: Explicit Top Level ACL Violation: Top-level declarations should specify Access Control Level keywords explicitly. (explicit_top_level_acl)
<nopath>:1:5: warning: Prefixed Top-Level Constant Violation: Top-level constants should be prefixed by `k`. (prefixed_toplevel_constant)
<nopath>:1:20: warning: Trailing Closure Violation: Trailing closure syntax should be used whenever possible. (trailing_closure)
<nopath>:1:1: warning: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
Done linting! Found 5 violations, 0 serious in 1 file.
